### PR TITLE
Close CUDA 12 migration (redux)

### LINKED ISSUE #5390: CUDA 12 version in cbc.yaml - plans?
### Comment:

Just wondering what the plans/current thinking is for including cuda v12 here? Has there been a buildout discussed?

### PR DESCRIPTION
**Pull request by h-vetinari**
Since #5613 has been languishing for a while despite being essentially ready (aside from some open review requests to drop CUDA 12.4), let's actually get this over the finish line.

I made another clean-up push for the respective outstanding feedstocks. Below is the full list, but in any case, we already established that things are ready in #5613, so this just moves the needle even further into the "let's pull the trigger" area.

Closes #5390
Closes https://github.com/conda-forge/conda-forge.github.io/issues/1963

## Actively worked on

* [x] https://github.com/conda-forge/aihwkit-gpu-feedstock/pull/15
* [ ] https://github.com/conda-forge/ambertools-feedstock/pull/148
* [x] https://github.com/conda-forge/autoawq-feedstock/pull/13
* [x] https://github.com/conda-forge/cyrsoxs-feedstock/pull/19
* [x] https://github.com/conda-forge/faiss-split-feedstock/pull/75
* [x] https://github.com/conda-forge/graphvite-feedstock/pull/6
* [x] https://github.com/conda-forge/isce3-feedstock/pull/64
* [x] https://github.com/conda-forge/kaldi-feedstock/pull/48
* [x] https://github.com/conda-forge/kalpy-feedstock/pull/24
* [ ] https://github.com/conda-forge/molgrid-feedstock/pull/14
* [x] https://github.com/conda-forge/seekr2_openmm_plugin-feedstock/pull/21
* [x] https://github.com/conda-forge/tsnecuda-feedstock/pull/12

## Stagnant

* [ ] https://github.com/conda-forge/katago-feedstock/pull/9
* [ ] https://github.com/conda-forge/lammps-feedstock/pull/188
* [ ] https://github.com/conda-forge/meld-feedstock/pull/23
* [ ] https://github.com/conda-forge/prismatic_split-feedstock/pull/54
* [ ] https://github.com/conda-forge/rmm-feedstock/pull/31

## Blocked upstream

* https://github.com/conda-forge/fairscale-feedstock/pull/32 (not compatible with new pytorch)
* https://github.com/conda-forge/torchsparse-feedstock/pull/30 (not compatible with new pytorch)

## Dead feedstocks

* https://github.com/conda-forge/heavydb-ext-feedstock/pull/65
* https://github.com/conda-forge/hdk-feedstock/pull/56
* https://github.com/conda-forge/libtvm-feedstock/pull/40
* https://github.com/conda-forge/timemory-feedstock
* https://github.com/conda-forge/tvm-py-feedstock (by virtue of depending on dead parent `libtvm`)

### CODE REVIEW

**Comment by h-vetinari** [comment]
Ping @jakirkham; I'd like to close this and then start discussing about a CUDA 12.4 or 12.6 migration (some packages already need newer versions than 12.0). I think we're ready here (and have been for quite a while), but since you invested so much time & energy on this migration I don't want to move forward without your input.

**Comment by jakirkham** [comment]
Hey Axel, sorry for the slow reply here

Axel and I discussed this offline after the last conda-forge meeting as we didn't get to this item in the meeting

Generally we agreed that the CUDA 12.0 migration has reached completion. There were few points I suggested we work on:

1. Try to get PRs close to completion through
    1. As CUDA 12 breaks the packaging format relative to CUDA 11 or earlier, thought we should try and get as many feedstocks on the other side of that break as possible
    2. Over the past few weeks Axel and I have made a best effort at this. The remainder are either unmaintained or will require substantial upstream work to adopt. So this is done
2. Clean up migrator vendoring needed to workaround bugs seen in pybind11 using feedstocks
    1. The combination of CUDA 12.0 + GCC 12 + pybind11 was problematic. We found rolling back to GCC 11 worked around these issues. So vendored migrators in feedstocks to use the older GCC
    2. Since then pybind11 included a fix in 2.12.0, which addressed this issue
    3. Have worked with maintainers to clean these up anywhere they remained. More context in comment: https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/5340#discussion_r1737899653
     4. So this is also done
3. Try to clear awaiting parents
    1. This isn't a hard blocker, but same as 1 we wanted to get this reasonably closed out due to the breaking nature of CUDA 12
    2. There are only 2 feedstocks without PRs. One appears unmaintained (like its parent). The other has an error as the feedstock is unmaintained and has fallen far behind
    3. In other words this is done

According to the status page 98% of PRs are made. Only 12 PRs remain open. With all of the work done above (thank you Axel for your hard work! 🙏), think we can safely declare this done

**Review by jakirkham — APPROVED** [review]

**Comment by jakirkham** [comment]
Thanks Axel! 🙏

**Comment by jakirkham** [comment]
Am noticing an issue with `zip_keys`: https://github.com/conda-forge/ucx-split-feedstock/pull/195#issuecomment-2323111570

Axel, could you please take a quick look to make sure we are not missing anything here?

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -7,6 +7,7 @@ c_compiler_version:            # [unix]
   - 12                         # [linux]
   - 16                         # [osx]
   - 11                         # [os.environ.get("CF_CUDA_ENABLED", "False") == "True" and linux]
+  - 12                         # [os.environ.get("CF_CUDA_ENABLED", "false") == "true" and linux]
```
Comment (h-vetinari) [inline recipe/conda_build_config.yaml:10]: I overlooked one of the places when taking over https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/5799/commits/1cf8857b12906e16641423e5572d5ae0b164e2ae where you had put in lower-case booleans, sorry.

Fixed in https://github.com/conda-forge/conda-forge-pinning-feedstock/commit/23be5a0477d40f737eb6d15423713b277ec0788c

Comment (jakirkham) [reply recipe/conda_build_config.yaml:10]: No worries. Thanks Axel! 🙏

**Pull request merged**